lib/c/libintvector.h

```diff
@@ -3,28 +3,35 @@
 
 #include <sys/types.h>
 
-// # DEBUGGING FLAGS
-// =================
-// It is possible to debug the trace of the primitives defined in
-// this file by using the [DEBUG_VECTOR_TRACE] C flag.
-// As we use the same vector types to manipulate blocks of uint32 and blocks
-// of uint64, the log results will vary with the endianess, in particular for
-// some generic operations like [and] or [xor]. By default, the printing is
-// performed as if we were manipulating blocks of uint32. If you want to
-// switch to blocks of uint64, use the flag: [DEBUG_VECTOR_TRACE_ELEMENTS_64].
-// Note that if those flags are activated, it may be necessary to tweak a bit
-// the compilation options to build HACL. More specifically, you may need to
-// always activate the compiler options to use vector support (even for files
-// which actually don't make use of vectors, if they have libintvector.h as
-// a dependency). When comparing traces, note that some instructions are not
-// compiled in the same order on the different platforms, but it doesn't lead
-// to a lot of discrepancies in practice.
+/* We include config.h here to ensure that the various feature-flags are
+ * properly brought into scope. Users can either run the configure script, or
+ * write a config.h themselves and put it under version control. */
+#if __has_include("config.h")
+#include "config.h"
+#endif
+
+/* # DEBUGGING FLAGS
+ * =================
+ * It is possible to debug the trace of the primitives defined in
+ * this file by using the [DEBUG_VECTOR_TRACE] C flag.
+ * As we use the same vector types to manipulate blocks of uint32 and blocks
+ * of uint64, the log results will vary with the endianess, in particular for
+ * some generic operations like [and] or [xor]. By default, the printing is
+ * performed as if we were manipulating blocks of uint32. If you want to
+ * switch to blocks of uint64, use the flag: [DEBUG_VECTOR_TRACE_ELEMENTS_64].
+ * Note that if those flags are activated, it may be necessary to tweak a bit
+ * the compilation options to build HACL. More specifically, you may need to
+ * always activate the compiler options to use vector support (even for files
+ * which actually don't make use of vectors, if they have libintvector.h as
+ * a dependency). When comparing traces, note that some instructions are not
+ * compiled in the same order on the different platforms, but it doesn't lead
+ * to a lot of discrepancies in practice. */
 
 #define Lib_IntVector_Intrinsics_bit_mask64(x) -((x) & 1)
 
 #if defined(__x86_64__) || defined(_M_X64)
 
-// The following functions are only available on machines that support Intel AVX
+#if defined(EVERCRYPT_CAN_COMPILE_VEC128)
```

Comment:
This said AVX support before and now vec128 support.
Looking at the `detect_uint128` in `configure` that sets `EVERCRYPT_CAN_COMPILE_VEC128` it checks whether `__int128` is available or not. But we are using `__m128i` here, which is available starting with SSE2.
So I'm wondering if the check is correct and the naming is precise (enough).

Reply:
That's my fault I forgot to refresh the snapshots. The diff for `dist/configure` should be correct (I fixed that earlier mistake by Son in one of my commits but forgot to propagate `dist/configure`-->`dist/*/configure`)

Reply:
everything should be refreshed... thanks for catching this!